src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,62 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * A kind of tasks that need to be done before a specific date/time.
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate byDate;
+
+    /**
+     * Default constructor for Deadline.
+     *
+     * @param description the description of the task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        try {
+            byDate = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
```

Comment:
Should you add an error message for DateTimeParseException to give more informative feedback to user?